models/models.go

```diff
@@ -626,93 +612,40 @@ func (m *RouteManager) checkDistribution(r *Route) bool {
 	return *dist.Status == "Deployed" && *dist.DistributionConfig.Enabled
 }
 
-func (m *RouteManager) solveChallenges(clients map[acme.Challenge]*acme.Client, challenges []acme.AuthorizationResource) map[string]error {
-	errs := make(chan map[string]error)
-
-	for _, client := range clients {
-		go func(client *acme.Client) {
-			errs <- client.SolveChallenges(challenges)
-		}(client)
-	}
-
-	var failures map[string]error
-	for challenge, _ := range clients {
-		failures = <-errs
-		m.logger.Info("solve-challenges", lager.Data{
-			"challenge": challenge,
-			"failures":  failures,
-		})
-		if len(failures) == 0 {
-			return failures
-		}
-	}
-
-	return failures
-}
-
-func (m *RouteManager) deployCertificate(route Route, cert acme.CertificateResource) error {
-	expires, err := acme.GetPEMCertExpiration(cert.Certificate)
+func (m *RouteManager) deployCertificate(route Route, certResource certificate.Resource) error {
+	cert, err := certcrypto.ParsePEMCertificate(certResource.Certificate)
 	if err != nil {
 		return err
 	}
+	expires := cert.NotAfter
 
 	name := fmt.Sprintf("cdn-route-%s-%s", route.InstanceId, expires.Format("2006-01-02_15-04-05"))
 
 	m.logger.Info("Uploading certificate to IAM", lager.Data{"name": name})
 
-	certId, err := m.iam.UploadCertificate(name, cert)
+	certId, err := m.iam.UploadCertificate(name, certResource)
 	if err != nil {
 		return err
 	}
 
 	return m.cloudFront.SetCertificateAndCname(route.DistId, certId, route.GetDomains())
 }
 
-func (m *RouteManager) ensureChallenges(route *Route, client *acme.Client, update bool) error {
-	if len(route.ChallengeJSON) == 0 {
-		challenges, errs := client.GetChallenges(route.GetDomains())
-		if len(errs) > 0 {
-			return fmt.Errorf("Error(s) getting challenges: %v", errs)
-		}
+// todo (mxplusb): find a better way to do this.
+func (m *RouteManager) GetDNSInstructions(route *Route) ([]string, error) {
+	// we need to set this to a length of 3 so we can put the instructions in the proper place.
+	instructions := make([]string, 3)
 
-		var err error
-		route.ChallengeJSON, err = json.Marshal(challenges)
-		if err != nil {
-			return err
+	for key := range m.dnsPresenter {
+		if strings.Contains(key, "domain") {
+			instructions[0] = fmt.Sprintf("name: %s", strings.Split(key, "domain")[0])
 		}
-
-		if update {
-			return m.db.Save(route).Error
+		if strings.Contains(key, "keyauth") {
+			instructions[1] = fmt.Sprintf("value: %s", strings.Split(key, "keyauth")[0])
 		}
-		return nil
 	}
 
-	return nil
-}
-
-func (m *RouteManager) GetDNSInstructions(route *Route) ([]string, error) {
-	var instructions []string
-	var challenges []acme.AuthorizationResource
-
-	user, err := route.loadUser(m.db)
-	if err != nil {
-		return instructions, err
-	}
+	instructions[2] = fmt.Sprintf("ttl: 120")
```

Comment:
do we know the ttl will always be 120? we were getting it from LetsEncrypt before.

Reply:
I wasn't able to figure this out in the new client, but it honestly doesn't matter. I generally set mine to 60 seconds and it works as intended.